src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerResponseToServiceApplication.java

```diff
@@ -0,0 +1,91 @@
+import static uk.gov.hmcts.divorce.divorcecase.model.AlternativeServiceType.DISPENSED;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AosOverdue;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingAos;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingBailiffReferral;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingServiceConsideration;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerResponseToServiceApplication implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_RESPONSE_TO_SERVICE_APPLICATION = "caseworker-response-to-service-application";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_RESPONSE_TO_SERVICE_APPLICATION)
+            .forStates(AwaitingAos, AosOverdue)
+            .name("Response to service app")
+            .description("Response to service application")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showSummary()
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
+            .grantHistoryOnly(LEGAL_ADVISOR, SOLICITOR))
+            .page("uploadDocument")
+            .pageLabel("Upload document")
+            .complex(CaseData::getDocuments)
+            .optional(CaseDocuments::getDocumentsUploaded)
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
+        final CaseDetails<CaseData, State> details,
+        final CaseDetails<CaseData, State> beforeDetails
+    ) {
+        log.info("Caseworker response to service application about to submit callback invoked");
+
+        List<String> validationErrors = new java.util.ArrayList<>(Collections.emptyList());
+
+        CaseData caseData = details.getData();
+        AlternativeServiceType alternativeServiceType = caseData.getAlternativeService().getAlternativeServiceType();
+
+        State state = details.getState();
+
+        if (DEEMED.equals(alternativeServiceType) || DISPENSED.equals(alternativeServiceType)) {
+            state = AwaitingServiceConsideration;
+        } else if (BAILIFF.equals(alternativeServiceType)) {
+            state = AwaitingBailiffReferral;
+        } else {
+            validationErrors.add("alternate service type doesn't match any of the required options");
```

Comment:
this message might be seen by caseworker so may be best to check what we need to display